api/workload/v2/stream.go

```diff
@@ -139,23 +142,31 @@ func (c *streamManager) start() {
 	}
 }
 
-func (c *streamManager) newStream(ctx context.Context, addr string) (stream workload.SpiffeWorkloadAPI_FetchX509SVIDClient, closer io.Closer, err error) {
+func (c *streamManager) newStream(dialCtx, streamCtx context.Context, addr string) (*managedStream, error) {
 	backoff := newBackoff()
 	for {
-		conn, err := newConn(ctx, addr)
+		var stream workload.SpiffeWorkloadAPI_FetchX509SVIDClient
+		var err error
```

Comment:
Still redundant because of 

```
conn, err := newConn(ctx, addr)
```

afaict

Reply:
defining these here so that the `goto` doesn't skip over instantiations of new variables. it's kind of jank, maybe i'll rewrite the retries to be a wrapper func rather than a for loop like this.

Reply:
Oh that's subtle, I would fully support a rewrite that avoids the goto.

Reply:
+1